src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -205,72 +206,125 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
             return path;
         }
 
-        public static string LookupTargetFrameworkFromProjectFile(string projectLocation)
-        {
-            var projectFile = FindProjectFileInDirectory(projectLocation);
 
-            var xdoc = XDocument.Load(projectFile);
-
-            var element = xdoc.XPathSelectElement("//PropertyGroup/TargetFramework");
-            return element?.Value;
-        }
-
-        /// <summary>
-        /// Retrieve the `OutputType` property of a given project
+        // <summary>
+        /// Looks up specified properties from a project file.
         /// </summary>
-        /// <param name="projectLocation">Path of the project</param>
-        /// <returns>The value of the `OutputType` property</returns>
-        public static string LookupOutputTypeFromProjectFile(string projectLocation)
+        /// <param name="projectLocation">The location of the project file.</param>
+        /// <param name="propertyNames">The names of the properties to look up.</param>
+        /// <returns>A dictionary of property names and their values.</returns>        
+        public static Dictionary<string, string> LookupProjectProperties(string projectLocation, params string[] propertyNames)
         {
-            var process = new Process();
-            var output = string.Empty;
-            var msbuildProcessFailed = false;
-            try
+            var projectFile = FindProjectFileInDirectory(projectLocation);
+            var properties = new Dictionary<string, string>();
+            var arguments = new List<string>
             {
-                process.StartInfo = new ProcessStartInfo()
+                "msbuild",
+                projectFile,
+                "-nologo",
+                $"--getProperty:{string.Join(',', propertyNames)}"
+            };
+
+            var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
                 {
                     FileName = "dotnet",
-                    Arguments = $"msbuild {projectLocation} -getProperty:OutputType",
+                    Arguments = string.Join(" ", arguments),
                     RedirectStandardOutput = true,
+                    RedirectStandardError = true,
                     UseShellExecute = false,
-                    WindowStyle = ProcessWindowStyle.Hidden
-                };
-
+                    CreateNoWindow = true
+                }
+            };
+            try
+            {
                 process.Start();
-                output = process.StandardOutput.ReadToEnd();
-                var hasExited = process.WaitForExit(5000);
+                string output = process.StandardOutput.ReadToEnd();
+                string error = process.StandardError.ReadToEnd();
+                process.WaitForExit();
```

Comment:
Should we keep the 5 second timeout or at least some timeout? I'm thinking of @philasmar recent PR that at least added `int.MaxValue` to avoid deadlocks in this API. https://github.com/aws/aws-extensions-for-dotnet-cli/pull/342

Reply:
sounds good i will update to 5 seconds